src/core/Table/Table.tsx

```diff
@@ -642,6 +694,21 @@ export const Table = <
                       {...columnProps}
                       key={columnProps.key}
                       title={undefined}
+                      draggable={props.enableDraggableColumns}
+                      onDragStart={(event) =>
+                        enableDraggableColumns &&
+                        dragStartHandler(event, column.id)
+                      }
+                      onDragOver={
```

Comment:
I think we should check position like whether it is first half of the column or the last half and to add column there accordingly:
![tree-column-reorder](https://user-images.githubusercontent.com/36186912/155743972-ad118a6a-3a50-4464-af74-2e2f53727ad3.gif)
This is what I did:
![image](https://user-images.githubusercontent.com/36186912/155744248-f8d09b00-806b-442d-a826-93b4b077903c.png)

Reply:
Also, you can use newly added CSS for columns borders: https://github.com/iTwin/iTwinUI/pull/498

Reply:
@bentleyvk Not sure how to modify cell classes: via event.currentTarget.classList or dedupe version from classNames. Also, would it be a performance issue to reassign same class multiple times in allowDrop event? I'm sorry but I think it would be better for you to look at it. "Allow edits by maintainers" is enabled. @itwin/itwinui-css is updated.